core/src/tasks/export/send.ts

```diff
@@ -16,31 +19,37 @@ export class ExportSend extends RetryableTask {
     };
   }
 
-  async runWithinTransaction(params) {
-    const destination = await Destination.scope(null).findOne({
-      where: { id: params.destinationId },
+  async run(params) {
+    let response: AsyncReturnType<typeof DestinationOps["sendExport"]>;
+    let app: App;
+
+    await CLS.wrap(async () => {
+      const destination = await Destination.scope(null).findOne({
+        where: { id: params.destinationId },
+      });
+      if (!destination) return;
+      app = await destination.$get("app");
+      const _export = await Export.findOne({
+        where: { id: params.exportId },
+      });
+      if (!_export) return; // the export was deleted
+      if (_export.completedAt) return; // be sure not to export twice
+
+      response = await destination.sendExport(_export);
     });
-    if (!destination) return;
-    const _export = await Export.findOne({ where: { id: params.exportId } });
-    if (!_export) return; // the export was deleted
-    if (_export.completedAt) return; // be sure not to export twice
 
-    const { success, retryDelay, error } = await destination.sendExport(
-      _export
-    );
+    if (!response) return; // we exited early
 
-    if (!success) {
-      if (retryDelay) {
-        const app = await destination.$get("app");
+    if (!response.success) {
+      if (response.retryDelay) {
         return CLS.enqueueTaskIn(
-          retryDelay,
+          response.retryDelay,
           "export:send",
           params,
           `exports:${app.type}`
         );
       } else {
-        // auto retry
-        throw error;
```

Comment:
This was the problem - if the plugin threw an error (vs returning errors), we'd get here and roll everything back.